test/net-test.js

```diff
@@ -0,0 +1,2218 @@
+          called = true;
+        };
+
+        await pool.handleGetData(peer, pkt);
+
+        assert(called);
+      });
+    });
+
+    describe('handleGetHeaders', function() {
+      function mockHash(height) {
+        const hash = Buffer.alloc(32, 0x00);
+        hash.writeUInt32LE(height);
+        return hash;
+      }
+
+      const network = Network.get('regtest');
+      let pool = null;
+
+      before(async () => {
+        pool = new Pool({
+          logger: Logger.global,
+          chain: {
+            network,
+            options: {checkpoints: true},
+            on: () => {},
+            synced: true,
+            findLocator: (locators) => {
+              assert.bufferEqual(locators[0], mockHash(0));
+
+              return locators[0];
+            },
+            getNextHash: (hash) => {
+              assert.bufferEqual(hash, mockHash(0));
+
+              return mockHash(1);
+            },
+            getEntry: (hash) => {
+              assert.bufferEqual(hash, mockHash(1));
+
+              return new ChainEntry({
+                version: 1,
+                hash: mockHash(1),
+                prevBlock: mockHash(0),
+                merkleRoot: Buffer.alloc(32, 0x00),
+                time: 1558629632,
+                bits: 486604799,
+                nonce: 10,
+                height: 1
+              });
+            },
+            getNext: (entry) => {
+              const height = entry.height + 1;
+
+              return new ChainEntry({
+                version: 1,
+                hash: mockHash(height),
+                prevBlock: entry.hash,
+                merkleRoot: Buffer.alloc(32, 0x00),
+                time: 1558629632,
+                bits: 486604799,
+                nonce: 10,
+                height: height
+              });
+            }
+          }
+        });
+      });
+
+      it('will send max headers from chain', async () => {
+        const peer = Peer.fromOptions(pool.options);
+        peer.handshake = true;
+
+        const locators = [mockHash(0)];
+        const stop = mockHash(7500);
+
+        const pkt = new packets.GetHeadersPacket(locators, stop);
+
+        let called = false;
+        peer.send = (packet) => {
+          assert.equal(packet.type, packets.types.HEADERS);
+          assert.equal(packet.items.length, 2000);
+          called = true;
+        };
+
+        await pool.handleGetHeaders(peer, pkt);
+        assert(called);
+      });
+
+      it('will continue until stop point', async () => {
+        const peer = Peer.fromOptions(pool.options);
+        peer.handshake = true;
+
+        const locators = [mockHash(0)];
+        const stop = mockHash(1500);
+
+        const pkt = new packets.GetHeadersPacket(locators, stop);
+
+        let called = false;
+        peer.send = (packet) => {
+          assert.equal(packet.type, packets.types.HEADERS);
+          assert.equal(packet.items.length, 1500);
```

Comment:
Should we also assert here that the last headers sent is in fact the `stop` (by hash)?